tensorly/decomposition/candecomp_parafac.py

```diff
@@ -197,8 +207,10 @@ def parafac(tensor, rank, n_iter_max=100, init='svd', svd='numpy_svd', tol=1e-8,
             for i, factor in enumerate(factors):
                 if i != mode:
                     pseudo_inverse = pseudo_inverse*tl.dot(tl.conj(tl.transpose(factor)), factor)
-            
-            #factor = tl.dot(unfold(tensor, mode), khatri_rao(factors, skip_matrix=mode).conj())
+
+            if mask is not None:
+                tensor = tensor*mask + tl.kruskal_to_tensor(factors, mask=1-mask)
```

Comment:
Yeah, here. `kruskal_to_tensor` will be the full dense shape of `tensor` (`factors` are dense in general).